caesar_cipher.rb

```diff
@@ -1,25 +1,24 @@
 def caesar_cipher(str, shift) 
-    lowercase_letters = Array('a'..'z')
-    uppercase_letters = Array('A'..'Z')
-    new_str = ''
-    str.each_char do |c|
-        if lowercase_letters.include?(c)
-            shifted_idx = lowercase_letters.index(c) + shift
-            # if the new idx is out of bounds, loop back to beginning for idx
-            # take idx value and subtract from 26 to get new idx value
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += lowercase_letters[shifted_idx]
-        elsif uppercase_letters.include?(c)
-            shifted_idx = uppercase_letters.index(c) + shift
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += uppercase_letters[shifted_idx]       
-        else
-            new_str += c       
-        end
+  lowercase_letters = Array('a'..'z')
+  uppercase_letters = Array('A'..'Z')
+  new_str = ''
+  str.each_char do |c|
```

Comment:
Take a look at `each_with_object` for this piece. It's the way to go when you're building up something (a string, a hash, counting things, etc.) from the elements in an array.

Reply:
When I try using each_with_object for the string, I get an error: 

```ruby
   undefined method `each_with_object' for "abc":String
```

Do I have to convert the string into an array first? And then do something like 
```ruby
  str_arry.each_with_object("") do |c, new_str|
     if condition
      new_str += ... 
```

Reply:
That would work. You can check out `split` to create an array of characters.

Reply:
So ... `str.split("").each_with_object("")  ... `